app/controllers/charts_controller.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+class ChartsController < ApplicationController
+  before_action :set_user_people, only: %i[index]
+  before_action :set_user_categories, only: %i[index]
+
+  # rubocop:disable Metrics/AbcSize
+  def index
+    if validate_date(params)
+      @start_date = Time.zone.today.beginning_of_month
+      @end_date = format_date(Time.zone.today)
+    else
+      @start_date = DateTime.parse(params[:search][:start_date])
+      @end_date = format_date(DateTime.parse(params[:search][:end_date]))
+    end
+
+    set_credits
+    set_debits
+  end
+  # rubocop:enable Metrics/AbcSize
+
+  private
+
+  def get_transactions(start_d, end_d, credit = true)
+    transaction = []
+
+    categories = credit ? Category.credit_cats(@categories) : Category.debit_cats(@categories)
+
+    categories.each do |category|
+      transactions = MoneyTransaction.where(person_category_id: category.person_categories,
+                                            created_at: start_d..end_d).sum(&:amount_value)
+      transaction += [[category.name, transactions]] if transactions.positive?
+    end
+    transaction
+  end
+
+  def format_date(date)
+    date.to_datetime.end_of_day
+  end
+
+  def set_user_people
+    @people = current_user.people
+  end
+
+  def set_user_categories
+    @categories = current_user.people.collect(&:categories).flatten.uniq
+  end
+
+  def set_credits
```

Comment:
methods `set_credits` and `set_debits` so needed?

Reply:
Fixed